examples/decomposition/plot_parafac2.py

```diff
@@ -0,0 +1,232 @@
+# the *factor matrices* and the *projection matrices*. The weights are similar
+# to the output of a CP decomposition. The factor matrices and projection 
+# matrices are somewhat different. For a CP decomposition, we only have the
+# weights and the factor matrices. However, since the PARAFAC2 factor matrices
+# for the second mode is given by
+# 
+# .. math::
+#     B_i = P_i B,
+# 
+# where :math:`B` is an :math:`R \times R` matrix and :math:`P_i` is an 
+# :math:`I \times R` projection matrix, we cannot store the factor matrices
+# the same as for a CP decomposition.
+# 
+# Instead, we store the factor matrix along the first mode (:math:`A`), the 
+# "blueprint" matrix for the second mode (:math:`B`) and the factor matrix 
+# along the third mode (:math:`C`) in one tuple and the projection matrices,
+# :math:`P_i`, in a separate tuple.
+# 
+# If we wish to extract the informative :math:`B_i` factor matrices, then we
+# use the ``tensorly.parafac2_tensor.apply_projection_matrices`` function on 
+# the PARAFAC2 tensor instance to get another wrapper object for two
+# variables: *weights* and *factor matrices*. However, now, the second element
+# of the factor matrices tuple is now a list of factor matrices, one for each
+# frontal slice of the tensor.
+# 
+# Likewise, if we wish to construct the tensor or the frontal slices, then we
+# can use the ``tensorly.parafac2_tensor.parafac2_to_tensor`` function. If the
+# decomposed dataset consisted of uneven-length frontal slices, then we can
+# use the ``tensorly.parafac2_tensor.parafac2_to_slices`` function to get a 
+# list of frontal slices.
+
+
+
+est_tensor = tl.parafac2_tensor.parafac2_to_tensor(decomposition)
+est_weights, (est_A, est_B, est_C) = tl.parafac2_tensor.apply_parafac2_projections(decomposition)
+
+##############################################################################
+# Compute performance metrics
+# ---------------------------
+
+
+reconstruction_error = la.norm(est_tensor - tensor)
+recovery_rate = 1 - reconstruction_error/la.norm(tensor)
+
+print(f'{recovery_rate:2.0%} of the data is explained by the model, which is expected with noise rate: {noise_rate}')
+
+
+# To evaluate how well the original structure is recovered, we calculate the tucker congruence coefficient.
+
+est_A, est_projected_Bs, est_C = tl.parafac2_tensor.apply_parafac2_projections(decomposition)[1]
+
+sign = np.sign(est_A)
+est_A = np.abs(est_A)
+est_projected_Bs = sign[:, np.newaxis]*est_projected_Bs
+
+est_A_normalised = est_A/la.norm(est_A, axis=0)
+est_Bs_normalised = [est_B/la.norm(est_B, axis=0) for est_B in est_projected_Bs]
+est_C_normalised = est_C/la.norm(est_C, axis=0)
+
+B_corr = np.array(est_Bs_normalised).reshape(-1, true_rank).T@np.array(Bs_normalised).reshape(-1, true_rank)/len(est_Bs_normalised)
+A_corr = est_A_normalised.T@A_normalised
+C_corr = est_C_normalised.T@C_normalised
+
+congruence_coefficient = np.mean(np.max(A_corr*B_corr*C_corr, axis=0))
```

Comment:
A quick comment on the way you find the optimal permutation. 
You compute (A'A*B'B*C*C), which is a square matrix of "scores", and  you expect the highest scores (except on the diagonal) to give you the permutations.
Actually, you may have in practice a component in A' that is the closest to all those in A, and then what you obtain from np.argmax is not a permutation?
A "clean" way to solve the assignment problem that you have is to use the munkres algorithm (linear assignment problem). An implementation is available in scikit-learn or in scipy.

Reply:
We updated the example so we now use`scipy.optimize.linear_sum_assignment` to find the optimal permutation of the factors.

Reply:
I would like to keep this version somewhere as an example that will work with all backends (without needing to convert to NumPy) -- we can add later in another small PR :)  
We can add a disclaimer about the potential issue.